catroid/src/main/java/org/catrobat/catroid/cast/CastManager.java

```diff
@@ -0,0 +1,459 @@
+						.inflate(R.layout.cast_pause_screen, null);
+				remoteLayout.addView(pausedView);
+				RelativeLayout.LayoutParams layoutParams = (RelativeLayout.LayoutParams) pausedView.getLayoutParams();
+				Project p = ProjectManager.getInstance().getCurrentProject();
+				layoutParams.height = p.getXmlHeader().getVirtualScreenHeight();
+				layoutParams.width = p.getXmlHeader().getVirtualScreenWidth();
+				pausedView.setLayoutParams(layoutParams);
+				pausedView.setBackgroundColor(0x66000000);
+				pausedScreenShowing = true;
+			}
+			pausedView.setVisibility(View.VISIBLE);
+			pausedScreenShowing = true;
+		}
+	}
+
+	public synchronized void resumeRemoteLayoutFromPauseScreen() {
+		if (remoteLayout != null && pausedView != null) {
+			pausedView.setVisibility(View.GONE);
+			pausedScreenShowing = false;
+		}
+	}
+
+	public synchronized void onStageDestroyed() {
+		if (isConnected) {
+			setRemoteLayoutToIdleScreen(initializingActivity);
+		}
+		stageViewDisplayedOnCast = null;
+		pausedView = null;
+		pausedScreenShowing = false;
+	}
+
+	private class MyMediaRouterCallback extends MediaRouter.Callback {
+
+		private long lastConnectionTry;
+
+		@Override
+		public void onRouteAdded(MediaRouter router, MediaRouter.RouteInfo info) {
+			// Add route to list of discovered routes
+			synchronized (this) {
+				for (int i = 0; i < routeInfos.size(); i++) {
+					MediaRouter.RouteInfo routeInfo = routeInfos.get(i);
+					if (routeInfo.equals(info)) {
+						routeInfos.remove(i);
+					}
+				}
+				routeInfos.add(info);
+				if (castButton != null) {
+					castButton.setVisible(mediaRouter.isRouteAvailable(mediaRouteSelector, MediaRouter
+							.AVAILABILITY_FLAG_REQUIRE_MATCH));
+				}
+				deviceAdapter.notifyDataSetChanged();
+			}
+		}
+
+		@Override
+		public void onRouteRemoved(MediaRouter router, MediaRouter.RouteInfo info) {
+			// Remove route from list of routes
+			synchronized (this) {
+				for (int i = 0; i < routeInfos.size(); i++) {
+					MediaRouter.RouteInfo routeInfo = routeInfos.get(i);
+					if (routeInfo.equals(info)) {
+						routeInfos.remove(i);
+						if (castButton != null && routeInfos.size() == 0) {
+							castButton.setVisible(mediaRouter.isRouteAvailable(mediaRouteSelector, MediaRouter
+									.AVAILABILITY_FLAG_REQUIRE_MATCH));
+						}
+						deviceAdapter.notifyDataSetChanged();
+					}
+				}
+			}
+		}
+
+		@Override
+		public void onRouteSelected(MediaRouter router, MediaRouter.RouteInfo info) {
+			synchronized (this) {
+				selectedDevice = CastDevice.getFromBundle(info.getExtras());
+				startCastService(initializingActivity);
+				lastConnectionTry = System.currentTimeMillis();
+				// Show a msg if still connecting after CAST_CONNECTION_TIMEOUT milliseconds
+				// and abort connection.
+				(new Handler()).postDelayed(new Runnable() {
+					@Override
+					public void run() {
+						synchronized (this) {
+							if (currentlyConnecting() && CastRemoteDisplayLocalService.getInstance() != null
```

Comment:
please put into a boolean variable describing what should be checked here (e.g. isCastAvailable).